File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.go
func findMaxAverage(nums []int, k int) float64 {
    queue := nums[:k]
    qTotal := sum(queue)
    maxTotal := qTotal
    for _, v := range nums[k:] {
        qTotal = qTotal - queue[0] + v
        queue = append(queue[1:], v)
        maxTotal = max(maxTotal, qTotal)
    }
    fmt.Println(maxTotal)
    return float64(maxTotal) / float64(k)
}

func sum(s []int) int {
    total := 0
    for _, n := range s {
        total += n
    }
    return total
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.c
double findMaxAverage(int* nums, int numsSize, int k){
    int total, max_total, lo, hi;
    
    total = 0;
    for (int i = 0; i < k; i++)
        total += nums[i];
    
    max_total = total;
    
    lo = 0;
    hi = k;
    while (hi < numsSize) {
        total = total - nums[lo++] + nums[hi++];
        max_total = (max_total >= total) ? max_total : total;
    }
    return (double)max_total / k;
}